### PR TITLE
Added GraphModel

### PR DESCRIPTION
**Pull request by Simcon**
Generally lifted from OrigoDB:

https://raw.githubusercontent.com/DevrexLabs/OrigoDB/dev/src/OrigoDB.Core/Modeling/GraphModel.cs

Added tests including end-to-end smoke tests covering basic usage and proving journaling/replaying of graph.

Aware this is an unsolicited PR so won't feel offended if not required or fits your roadmap!

### CODE REVIEW

File: src/Memstate.Core/Models/Graph/GraphModel.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        /// <summary>
+        /// Unique id generator, shared by nodes and edges
+        /// </summary>
+        private long _lastId;
+
+        //Nodes and edges
+        private readonly SortedDictionary<long, Node> _nodesById;
+        private readonly SortedDictionary<long, Edge> _edgesById;
+
+        [NonSerialized]
```
Comment (rofr) [inline src/Memstate.Core/Models/Graph/GraphModel.cs]: We haven't added support for snapshots yet so NonSerialized could be removed for now.

**Comment by Simcon** [comment]
Hi Robert. Thanks for the positive feedback! I'll resolve these changes some point this week/weekend. Looks like CLA has approved me as a different user so let me straighten that out well and we'll be good to go.

**Comment by Simcon** [comment]
I've pushed the requested changes and double-checked I've signed the CLA but some reason the status is still showing as unsigned?!

**Comment by rofr** [comment]
There's a known issue when using different email addresses for commits and the GH account itself:
https://github.com/cla-assistant/cla-assistant/issues/352#issuecomment-404404707

with a possible resolution proposed:
https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user

**Comment by Simcon** [comment]
Had to do a bit of a dance with GitHub but all seems good now.

**Comment by rofr** [comment]
Great, thanks for the contribution. If you'd like to have a go at some other bits let's have a chat on https://gitter.im/memstate/lobby

**Pull request merged**